plaso/parsers/shared/unified_logging/tracev3.py

```diff
@@ -0,0 +1,1221 @@
+              raw_data, 1, self._GetDataTypeMap('sockaddr'))
+          # IPv6
+          if sockaddr.family == 0x1E:
+            if raw_data[8:24] == b'\x00' * 16:
+              chars = 'IN6ADDR_ANY'
+            else:
+              try:
+                chars = ipaddress.ip_address(sockaddr.ipv6_ip.ip).compressed
+              except ValueError:
+                pass
+          # IPv4
+          elif sockaddr.family == 0x02:
+            chars = self._FormatPackedIPv4Address(
+                sockaddr.ipv4_address.segments)
+            if sockaddr.ipv4_port:
+              chars += ':{0:d}'.format(sockaddr.ipv4_port)
+          else:
+            logger.error('Unknown Sockaddr Family: {}'.format(sockaddr.family))
+            output += 'Invalid specifier'
+            i += 1
+            continue
+        elif 'network:in_addr' in custom_specifier:
+          ip_addr = self._ReadStructureFromByteStream(
+              raw_data, 0, self._GetDataTypeMap('ipv4_address'))
+          chars = self._FormatPackedIPv4Address(ip_addr.segments)
+        elif 'network:in6_addr' in custom_specifier:
+          chars = ipaddress.ip_address(raw_data).compressed
+        elif 'location:SqliteResult' in custom_specifier:
+          code = sqlite.SQLiteResultCodeHelper.GetResult(
+              self._ReadStructureFromByteStream(raw_data, 0,
+                                                uint32_data_type_map))
+          if code:
+            chars += '"{0:s}"'.format(code)
+          else:
+            logger.error('Unknown SQLite Code: {0:X}'.format(raw_data[0]))
+            output += 'Invalid specifier'
+            i += 1
+            continue
+        elif 'location:_CLLocationManagerStateTrackerState' in custom_specifier:
+          (
+              state_tracker_structure,
+              extra_state_tracker_structure,
+          ) = location.LocationManagerStateTrackerParser().Parse(
+              data_size, raw_data
+          )
+          chars = str({
+              **state_tracker_structure,
+              **extra_state_tracker_structure
+          })
+        elif 'location:_CLClientManagerStateTrackerState' in custom_specifier:
+          chars = str(
+              location.LocationClientStateTrackerParser().Parse(raw_data)
+          )
+        elif 'mdns:dnshdr' in custom_specifier:
+          dns_parser = dns.DNS()
+          chars = dns_parser.ParseDNSHeader(raw_data)
+        elif 'mdns:rd.svcb' in custom_specifier:
+          dns_parser = dns.DNS()
+          chars = dns_parser.ParseDNSSVCB(raw_data)
+        elif 'mdnsresponder:domain_name' in custom_specifier:
+          chars = ''.join([
+              '.' if not chr(s).isprintable() else chr(s)
+              for s in raw_data.replace(b'\n', b'').replace(b'\t', b'').replace(
+                  b'\r', b'')
+          ])
+        elif 'mdnsresponder:ip_addr' in custom_specifier:
+          ip_type = self._ReadStructureFromByteStream(raw_data[0:], 0,
```

Comment:
Do you have example data of this? I have a 0xf2 value that consists of 16 bytes only, no obvious IP version in the data.

Reply:
this might be related to the `mask.hash` custom specifier